src/pages/index.js

```diff
@@ -3,95 +3,157 @@ import Card from "/src/components/Card.js";
 import FormValidator from "/src/components/FormValidator.js";
 import Section from "/src/components/Section.js";
 import UserInfo from "/src/components/UserInfo.js";
+import PopupWithConfirmation from "/src/components/PopupWithConfirmation.js";
 import PopupWithForm from "/src/components/PopupWithForm.js";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```